src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -151,26 +152,27 @@ protected Job asJob() {
     }
 
     private List<ParameterValue> completeDefaultParameters(List<ParameterValue> parameters, Job<?,?> project) {
-        List<ParameterValue> completeListOfParameters = Lists.newArrayList(parameters);
-        List<String> names = Lists.transform(parameters, new Function<ParameterValue, String>() {
-            @Override public String apply(ParameterValue input) {
-                return input.getName();
-            }
-        });
+        Map<String,ParameterValue> allParameters = new HashMap<>();
+        for (ParameterValue pv : parameters) {
+            allParameters.put(pv.getName(), pv);
+        }
         if (project != null) {
             ParametersDefinitionProperty pdp = project.getProperty(ParametersDefinitionProperty.class);
             if (pdp != null) {
                 for (ParameterDefinition pDef : pdp.getParameterDefinitions()) {
-                    if (!names.contains(pDef.getName())) {
+                    if (!allParameters.containsKey(pDef.getName())) {
                         ParameterValue defaultP = pDef.getDefaultParameterValue();
                         if (defaultP != null) {
-                            completeListOfParameters.add(defaultP);
+                            allParameters.put(defaultP.getName(), defaultP);
                         }
+                    } else {
+                        // Get the description of specified parameters here. UI submission of parameters uses formatted description.
+                        allParameters.get(pDef.getName()).setDescription(pDef.getFormattedDescription());
```

Comment:
That does not accord with what I see [here](https://github.com/jenkinsci/jenkins/blob/90f778d2819cfd223f8815a3d84ce3e904d91fb0/core/src/main/java/hudson/model/StringParameterDefinition.java#L129) (web submission) or [here](https://github.com/jenkinsci/jenkins/blob/90f778d2819cfd223f8815a3d84ce3e904d91fb0/core/src/main/java/hudson/model/StringParameterDefinition.java#L134) (`build` CLI command submission). And I see no test coverage of `description`s that would be transformed by a `MarkupFormatter`.

As of SECURITY-353. [`ParameterValue.description` is not assumed to have been formatted](https://github.com/jenkinsci/jenkins/blob/90f778d2819cfd223f8815a3d84ce3e904d91fb0/core/src/main/resources/hudson/model/StringParameterValue/value.jelly#L30).

Reply:
Works for me.